In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import re
import nltk
from nltk.stem import WordNetLemmatizer
from tensorflow.keras.preprocessing.text import one_hot
from tensorflow.keras.preprocessing.sequence import pad_sequences
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Embedding, LSTM, Bidirectional, Dense

In [21]:
df = pd.read_csv('../Data/Language Detection.csv')

In [22]:
df.head()

Text Language
0   Nature, in the broadest sense, is the natural...  English
1  "Nature" can refer to the phenomena of the phy...  English
2  The study of nature is a large, if not the onl...  English
3  Although humans are part of nature, human acti...  English
4  [1] The word nature is borrowed from the Old F...  English

In [23]:
df.tail()

Text Language
10332  ನಿಮ್ಮ ತಪ್ಪು ಏನು ಬಂದಿದೆಯೆಂದರೆ ಆ ದಿನದಿಂದ ನಿಮಗೆ ಒ...  Kannada
10333  ನಾರ್ಸಿಸಾ ತಾನು ಮೊದಲಿಗೆ ಹೆಣಗಾಡುತ್ತಿದ್ದ ಮಾರ್ಗಗಳನ್...  Kannada
10334  ಹೇಗೆ ' ನಾರ್ಸಿಸಿಸಮ್ ಈಗ ಮರಿಯನ್ ಅವರಿಗೆ ಸಂಭವಿಸಿದ ಎ...  Kannada
10335  ಅವಳು ಈಗ ಹೆಚ್ಚು ಚಿನ್ನದ ಬ್ರೆಡ್ ಬಯಸುವುದಿಲ್ಲ ಎಂದು ...  Kannada
10336  ಟೆರ್ರಿ ನೀವು ನಿಜವಾಗಿಯೂ ಆ ದೇವದೂತನಂತೆ ಸ್ವಲ್ಪ ಕಾಣು...  Kannada

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10337 entries, 0 to 10336
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Text      10337 non-null  object
 1   Language  10337 non-null  object
dtypes: object(2)
memory usage: 161.6+ KB


In [25]:
df.shape

(10337, 2)

In [26]:
df.columns

Index(['Text', 'Language'], dtype='object')

The dataset contains 10,337 rows with two columns: "text" and "language." Both columns have a data type of "object." The "text" column likely contains textual data, while the "language" column may store information about the language of the corresponding text. 

In [27]:
df.duplicated().sum()

66

In [28]:
df.drop_duplicates(inplace=True)

In [29]:
df.duplicated().sum()

0

 It was discovered that there were 66 rows with duplicate data. To address this issue, the decision was made to remove these duplicate rows, resulting in a refined dataset with 10,271 rows. 

In [30]:
df.sample(6)

Text Language
7927  Ayrıca size gerçekten akıcı akıllı kelime numa...  Turkish
9087  [78] بمرور الوقت، طورت ويكيبيديا عملية تسوية ن...   Arabic
365   [67] According to the rules on the English Wik...  English
9029  ونتيجة لذلك، يمكن أن تحتوي أي مقالة على معلوما...   Arabic
4064              si quelqu'un pleure et n'en veut pas.   French
5139  [240]​ En un informe de Hackernoon, sitio web ...  Spanish

In [31]:
df.isnull().sum()

Text        0
Language    0
dtype: int64

 it does not contain any null or missing values. This indicates that the dataset is complete and does not have any missing data.